### PR TITLE
Update release process to avoid staging week

### PR DESCRIPTION
**Pull request by mickaelistria**

### CODE REVIEW

File: devdoc/Releasing_Tycho.md [hunk 1]
```diff
@@ -1,11 +1,36 @@
...
 - [ ] Prepare the [release notes](https://github.com/eclipse/tycho/blob/master/RELEASE_NOTES.md) which should provide a quick overview of new features and bug fixes 
 - [ ] Create release record on https://projects.eclipse.org/projects/technology.tycho projects.eclipse.org , link the N&N to https://github.com/eclipse/tycho/blob/master/RELEASE_NOTES.md release notes]
-- [ ] Update versions using `mvn org.eclipse.tycho:tycho-versions-plugin:set-version -DnewVersion=<VERSION>` 
-- [ ] Commit version change, and create a git tag `tycho-<VERSION>` on this commit  
-- [ ] Update versions (same as above) to next `-SNAPSHOT` development version and push commit to `master` branch 
+- [ ] Create branch `tycho-N.M.x` (eg `tycho-2.4.x`) for upcoming release and push it to remote; this branch should remain frozen until the release, only major fixes for regressions could be merged in before release. Work can still happen regularly on the `master` branch.
+- [ ] Update versions on `master `to future release with `mvn org.eclipse.tycho:tycho-versions-plugin:set-version -DnewVersion=<NEXT_ERSION>-SNAPSHOT` and push to remote
```
Comment (Dufgui) [inline devdoc/Releasing_Tycho.md:9]:
```suggestion
- [ ] Update versions on `master `to future release with `mvn org.eclipse.tycho:tycho-versions-plugin:set-version -DnewVersion=<NEXT_VERSION>-SNAPSHOT` and push to remote 
```

File: devdoc/Releasing_Tycho.md [hunk 2]
```diff
@@ -1,11 +1,36 @@
...
-- [ ] Commit version change, and create a git tag `tycho-<VERSION>` on this commit  
-- [ ] Update versions (same as above) to next `-SNAPSHOT` development version and push commit to `master` branch 
+- [ ] Create branch `tycho-N.M.x` (eg `tycho-2.4.x`) for upcoming release and push it to remote; this branch should remain frozen until the release, only major fixes for regressions could be merged in before release. Work can still happen regularly on the `master` branch.
+- [ ] Update versions on `master `to future release with `mvn org.eclipse.tycho:tycho-versions-plugin:set-version -DnewVersion=<NEXT_ERSION>-SNAPSHOT` and push to remote 
+- [ ] Announce the intent to release and request feedback about snapshots on tycho-user@eclipse.org:
+```
+Subject: Please test snapshots for upcoming <VERSION> release
+
+We plan to release Tycho <VERSION> next week. For details of new features and bugfixes, see release notes [1].
+Please help by testing the SNAPSHOTSs build. To use it, change your tycho version to <VERSION>-SNAPSHOT and add snippet [2] to your pom.
```
Comment (Dufgui) [inline devdoc/Releasing_Tycho.md:15]:
```suggestion
Please help by testing the SNAPSHOTS build. To use it, change your tycho version to <VERSION>-SNAPSHOT and add snippet [2] to your pom.
```

File: devdoc/Releasing_Tycho.md [hunk 3]
```diff
@@ -22,44 +47,19 @@ This page describes the steps necessary to create releases of the Tycho project.
...
-
-... Wait until review date (usually a week later)...
-
-- [ ] After ~1 week of testing, promote the stage repository on https://oss.sonatype.org/
 - [ ] Wait for artifacts to be available on Maven central.
-- [ ] Push the release tags to git: `git push origin --tags`
-- [ ] Update https://github.com/eclipse/tycho/blob/master/RELEASE_NOTES.md to remove reference to staging repo
-- [ ] Merge the update to `org.eclipse.tycho-demo` and website in Gerrit
-- [ ] Announce the release
+- `git tag <TYCHO_VERSION>`
```
Comment (Dufgui) [inline devdoc/Releasing_Tycho.md:57]: maybe tag before sync to nexus ?

Comment (Dufgui) [reply devdoc/Releasing_Tycho.md:57]: is it synchronize with github release?  Where do you feed the content of the release in github? RELEASE_NOTES.md is enough to feed it ?

**Review by Dufgui — CHANGES_REQUESTED** [review]

File: devdoc/Releasing_Tycho.md [hunk 4]
```diff
@@ -22,44 +47,19 @@ This page describes the steps necessary to create releases of the Tycho project.
 </settings>
 </source>
 ```
-- [ ] Checkout the tag version `tycho-<VERSION>` so current commit is the tag.
+
+- [ ] Checkout branch `tycho-N.M.x`
+- [ ] Update version to remove `-SNAPSHOT` with `mvn org.eclipse.tycho:tycho-versions-plugin:set-version -DnewVersion=<VERSION>`
+- [ ] Update versions in tycho-demo folder
```
Comment (laeubi) [inline devdoc/Releasing_Tycho.md:53]: could this be done automatic or does this need manual adjustments?

File: devdoc/Releasing_Tycho.md (hunk 3 above)
Comment (laeubi) [reply devdoc/Releasing_Tycho.md:57]: Maybe creating the tag/release [as described here](https://docs.github.com/en/github/administering-a-repository/releasing-projects-on-github/managing-releases-in-a-repository)?

**Comment by HannesWell** [comment]
Maybe it is obvious, but a paragraph what to do in case major regressions are found could also be helpful. How to incorporate the changes to fix the regression? Probably cherry-pick the fixing commit from the master (merging would override the version and could add more changes then intended). And what happens with the release date then? Does it stay unchanged or is it postponed to one week after the time the regression was fixed. Is an E-Mail send to inform others about the fixed regression and the potentially moved release date?


Furthermore, following @laeubi's comment, I think the entire release process could be automated using GitHub actions or maybe also using Jenkins. For GitHub actions it should be possible to create two manually triggered workflows. One to prepare a release and one to perform a release, that correspondingly create the release-branch, send the e-mails and perform the version changes and tagging. The same could also be possible using multiple parametrized Jenkins jobs, but I'm not 100% certain about that.

File: devdoc/Releasing_Tycho.md (hunk 4 above)
Comment (mickaelistria) [reply devdoc/Releasing_Tycho.md:53]: It could probably be done automatically, but that's not in the scope of this change.

**Comment by mickaelistria** [comment]
The good thing about this document is that since it's now on GitHub, it becomes pretty easy to submit improvement. I'd like to change to keep focus on just delaying the staging phase; other improvements (usage of GitHub project management, automation...) would better be tracked separately.

**Pull request merged**